src/foraging_gui/Visualization.py

```diff
@@ -61,34 +73,43 @@ def _Update(self,GeneratedTrials=None,Channel=None):
 
         try:
             self._PlotBlockStructure()
+        except Exception as e:
+            logging.error(str(e))
+
+        try:
             self._PlotChoice()
+        except Exception as e:
+            logging.error(str(e))
+
+        try:
             self._PlotLicks()
         except Exception as e:
             logging.error(str(e))
+
         try:
             self._PlotMatching()
         except Exception as e:
             logging.error(str(e))
-        self.finish=1
+
     def _PlotBlockStructure(self):
```

Comment:
Why do you replace all ax2 (ax3, ax1) with self.ax2 (ax3, ax1)? Is there a specific consideration?

Reply:
because its simpler. There is no reason to do:
```
ax = self.ax1
ax.plot()
```
when we can use:
```
self.ax1.plot()
```